File: app/src/main/java/com/kota/Bahamut/Command/BahamutCommandLoadBlock.java
package com.kota.Bahamut.Command;

import androidx.annotation.NonNull;

import com.kota.Bahamut.ListPage.TelnetListPage;
import com.kota.Bahamut.ListPage.TelnetListPageBlock;
import com.kota.Telnet.Logic.ItemUtils;
import com.kota.Telnet.TelnetClient;

public class BahamutCommandLoadBlock extends TelnetCommand {
    int _block;

    public BahamutCommandLoadBlock(int aBlock) {
        this.Action = LoadBlock;
        this._block = aBlock;
    }

    public boolean containsArticle(int articleIndex) {
        return this._block == ItemUtils.getBlock(articleIndex);
    }

    public boolean isOperationCommand() {
        return false;
    }

    public void execute(TelnetListPage aListPage) {
        if (this._block >= 0) {
            TelnetClient.getClient().sendStringToServer(String.valueOf((this._block * 20) + 1));
        }
    }

    public void executeFinished(TelnetListPage aListPage, TelnetListPageBlock aPageData) {
        setDone(true);
    }

    @NonNull
    public String toString() {
        return "[LoadBlock][block=" + this._block + " targetIndex=" + ((this._block * 20) + 1) + "]";
    }
}

File: app/src/main/java/com/kota/Bahamut/Command/BahamutCommandLoadBlock.kt
package com.kota.Bahamut.Command

import com.kota.Bahamut.ListPage.TelnetListPage
import com.kota.Bahamut.ListPage.TelnetListPageBlock
import com.kota.Telnet.Logic.ItemUtils.getBlock
import com.kota.Telnet.TelnetClient

class BahamutCommandLoadBlock(aBlock: Int) : TelnetCommand() {
    var _block: Int

    init {
        this.Action = BahamutCommandDefs.Companion.LoadBlock
        this._block = aBlock
    }

    fun containsArticle(articleIndex: Int): Boolean {
        return this._block == getBlock(articleIndex)
    }

    override fun isOperationCommand(): Boolean {
        return false
    }

    override fun execute(aListPage: TelnetListPage?) {
        if (this._block >= 0) {
            TelnetClient.getClient().sendStringToServer(((this._block * 20) + 1).toString())
        }
    }

    override fun executeFinished(aListPage: TelnetListPage?, aPageData: TelnetListPageBlock?) {
        setDone(true)
    }

    override fun toString(): String {
        return "[LoadBlock][block=" + this._block + " targetIndex=" + ((this._block * 20) + 1) + "]"
    }
}
